service/misc_services.cc

```diff
@@ -184,14 +184,35 @@ future<lowres_clock::duration> cache_hitrate_calculator::recalculate_hitrates()
     }).then([this] {
         _slen = _gstate.size();
         using namespace std::chrono_literals;
-        return _gossiper.add_local_application_state(gms::application_state::CACHE_HITRATES, gms::versioned_value::cache_hitrates(_gstate)).then([this] {
-            // if max difference during this round is big schedule next recalculate earlier
-            if (_diff < 0.01) {
-                return lowres_clock::duration(2000ms);
-            } else {
-                return lowres_clock::duration(500ms);
-            }
-        });
+        auto now = lowres_clock::now();
+        // Publish CACHE_HITRATES in case:
+        //
+        // - We haven't published it at all
+        // - The diff is bigger enough and we haven't published in the last 5 seconds
+        //
+        // Note: A peer node can know the cache hitrate through read_data
+        // read_mutation_data and read_digest RPC verbs which have
+        // cache_temperature in the response. So there is no need to update
+        // CACHE_HITRATES through gossip in high frequency.
+        bool do_publish = (_published_nr == 0) || ( _diff > 0.01 && (now - _published_time) > 5000ms);
```

Comment:
The diff 0.01 = 1%  is not that big in my view. 

Maybe we can do something like this:

bool do_publish = (_published_nr == 0) || ( _diff > 0.01 && (now - _published_time) > 5000ms) || _diff > 0.1 ;

Since most of the cases, other nodes will not use the hit rate from gossip but from the read verbs, I want to reduce the update if possible.

Reply:
OK ;-)

Changed in v2.